### PR TITLE
Add Linux support

### PR DESCRIPTION
**Pull request by hedjour**
Hi,
Here is a proposal of linux support, I just have ssh error with curl in eduroam network...
I put all version software variable in an unique script.
I put also function in an unique script.
I transform the script calling to be able to run it whithout cd dir_of_script and ./name of script

### CODE REVIEW

File: install_abba.sh [hunk 1]
```diff
@@ -110,51 +88,8 @@ mkdir "$temp_dl_dir"
...
-	exit 1 # We cannot proceed
-elif [[ "$OSTYPE" == "darwin"* ]]; then
-	fiji_executable_file="Contents/MacOS/ImageJ-macosx"
-	fiji_url="https://downloads.imagej.net/fiji/latest/fiji-macosx.zip"
-elif [[ "$OSTYPE" == "msys" ]]; then
-	fiji_executable_file="ImageJ-win64.exe"
-	fiji_url="https://downloads.imagej.net/fiji/latest/fiji-win64.zip"
-fi
-
+."$scriptpath/install_fiji.sh" "$path_install"
```
Comment (NicoKiaru) [inline install_abba.sh:91]: 👍

File: install_abba.sh [hunk 2]
```diff
@@ -257,61 +194,12 @@ all_args="$argElastixPath,$argTransformixPath"
...
-		if [[ -f "$qupath_path" ]]; then
-			echo "QuPath successfully installed"
-		else
-			echo "QuPath installation failed, please retry with administrator rights or install in a folder requiring less priviledge"
-			pause "Press [Enter] to end the script"
-			exit 1 # We cannot proceed
-		fi
-	fi	
-fi
+$scriptpath/install_qupath.sh "$path_install"
```
Comment (NicoKiaru) [inline install_abba.sh:197]: 👍

File: install_abba.sh [hunk 3]
```diff
@@ -320,7 +208,7 @@ rm -r "$temp_dl_dir"
 
 if [[ "$OSTYPE" == "darwin"* ]]; then
 	echo "Your OS: Mac OSX, make the qupath extension folder not read only"
-	chmod -R a+w "$path_install/QuPath Common Data_0.3"
+	chmod -R a+w "$path_install/QuPath Common Data_0.4"
```
Comment (NicoKiaru) [inline install_abba.sh:211]: Thanks!

File: version_software_script.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+warpy_extension_version=0.2.0
+stardist_extension_version=0.4.0
+biop_omero_extension_version=0.3.2
+warpy_extension_version=0.2.0
+
+#QuPath arguments :
+argQuPathUserPath="defaultQuPathUserPath=\"$path_install/QuPath_Common_Data_0.4\""
+argQuPathPrefNode="quPathPrefsNode=\"io.github.qupath/0.4\""
+argQuPathExtensionURL="quPathExtensionURL=\"$qupath_abba_extension_url\""
+argQuitAfterInstall="quitAfterInstall=\"true\""
```
Comment (NicoKiaru) [inline version_software_script.sh:20]: Ah! Here they are!

**Review by NicoKiaru — COMMENTED** [review]
I have an issue on Windows when executing full_install.sh:

```
Fiji should now be up-to-date
------ QuPath BIOP Full Installer Script -------------
This batch file downloads and install QuPath + extensions on your computer
Your OS: Windows
------- Installation path validation
All components will be installed in:
C:/
bash: .C:/Users/chiarutt/Downloads/biop-bash-scripts-main(1)/biop-bash-scripts-main/install_qupath.sh: No such file or directory
bash: .C:/Users/chiarutt/Downloads/biop-bash-scripts-main(1)/biop-bash-scripts-main/install_qupath_extensions.sh: No such file or directory
------ ImageJ/Fiji BIOP Full Installer Script -------------
This batch file downloads and install all components on your computer
Your OS: Windows
------- Installation path validation
All components will be installed in:
C:/
------ BIOP Full Installer Script -------------
This batch file downloads and install Fiji, QuPath and associated Biop selected components on your computer
Your OS: Windows
------- Installation path validation
Please enter the installation path (windows: C:/, mac: /Applications/, , Linux : /home/user/abba) \n
        The directory must exist first.
Path:
```

The PR looks super great, avoiding a lot of duplication. Thanks a lot for your contribution!

I'll check tomorrow what is the cause of the issue, it's probably not much. However, currently, all the update sites installation take forever. To debug the issue, I will remove the update site install temporiraly to be able to do several fast rounds of testing.

File: full_install_biop_fiji.sh [hunk 5]
```diff
@@ -87,5 +69,5 @@ fi
 echo "All components will be installed in:"
 echo "$path_install"
 
-./install_fiji.sh "$path_install"
-./install_fiji_update_sites.sh "$path_install"
+. "$scriptpath/install_fiji.sh" "$path_install"
```
[the remark this answers is not in the document]
Comment (hedjour) [reply full_install_biop_fiji.sh:72]: Nop just an error my bad.
I will correct it

File: install_abba.sh [hunk 6]
```diff
@@ -32,33 +34,8 @@ while getopts ":h" option; do
...
-		done
-    eval $__resultvar="'$myresult'"
-}
-
 # ----------------- COMPONENTS VERSION -----------
-qupath_version=0.4.3
-abba_extension_version=0.1.4
-elastix_version=5.0.1
-
+source "$scriptpath/version_software_script.sh"
```
[the remark this answers is not in the document]
Comment (hedjour) [reply install_abba.sh:38]: Your welcome, I will continue to summarise code

File: install_abba.sh [hunk 7]
```diff
@@ -257,61 +194,12 @@ all_args="$argElastixPath,$argTransformixPath"
...
+$scriptpath/install_qupath.sh "$path_install"
 
 echo ------ Setting up QuPath extension ------
-
+#TODO refine here verify if QPath is in directory define a QuPath dir where the version is not precised
 # See https://imagej.net/scripting/headless to deal with the mess of single quotes vs double quotes
-argQuPathUserPath="defaultQuPathUserPath=\"$path_install/QuPath Common Data_0.3\""
-argQuPathPrefNode="quPathPrefsNode=\"io.github.qupath/0.3\""
-argQuPathExtensionURL="quPathExtensionURL=\"$qupath_abba_extension_url\""
-argQuitAfterInstall="quitAfterInstall=\"true\""
```
[the remark this answers is not in the document]
Comment (hedjour) [reply install_abba.sh:314]: Normaly I put it in the script version software script ?
Is it possible to don't have the version of QuPath in these variable.

**Comment by hedjour** [comment]
I resolv this morning a bug with QuPath script and tar commands.
Sorry I'm not very confident with PR say me if I need to make another, with our correction.

**Comment by NicoKiaru** [comment]
Ah, I've been working as well on your PR, but I can't push to your branch. Can you give me rights to push to your main branch on [hedjour/biop-bash-scripts](https://github.com/hedjour/biop-bash-scripts)? Otherwise I merge this PR, push my modifications, and you can re-check / modify.

I can also give you edit rights on this repo. Let me know what you prefer.

**Comment by hedjour** [comment]
I give you the right

**Comment by hedjour** [comment]
I have finish my test on linux all run OK.

**Comment by NicoKiaru** [comment]
Perfect, I'll test all on windows, and try to merge today.

Thanks a lot for your contribution! I'm glad you removed all that duplicated code 👍

**Comment by hedjour** [comment]
I'm trying now on windows for my colleagues. I just can try on mac

**Comment by NicoKiaru** [comment]
Yeah, we bought one mac in the facility. Its only role is to do such testings. I'll try to find the courage to boot it and run the scripts. It's super painful with all the security layers.

**Comment by hedjour** [comment]
Sorry for the many push I try to fix all bug on windows + GnuLinux

**Comment by NicoKiaru** [comment]
All good. I'll probably have time to finish only next week. I'm not touching it for now.

**Comment by hedjour** [comment]
Ok I let you known when I finished my tests

**Comment by NicoKiaru** [comment]
> I see a bug on windows of C:\c\Users... in groovy scriptcall

Ah! Sorry I missed this. I'll have a look tomorrow

**Comment by hedjour** [comment]
No problem if you need something do not hesitate.
Just a litle thing have you also really hard Fiji download.

**Comment by hedjour** [comment]
@NicoKiaru Thanks for the news no problem.

**Comment by NicoKiaru** [comment]
Still no time, at this point I won't make any prediction...

I also need to check whether the possibility to add several update sites at the same time works now (https://github.com/imagej/imagej-updater/commit/20afd7297cf53a0f093f781f8c2d18e20ea2ccdf).

FOrum post : https://forum.image.sc/t/repeated-imagej-updater-calls-cause-403-errors/76613

**Comment by NicoKiaru** [comment]
Hi @hedjour , can you remove all updates sites except PTBIOP and OMERO  from https://github.com/BIOP/biop-bash-scripts/blob/66dbb092a1987171837aa983d1325b36b1a354b9/install_fiji_update_sites.sh#L136-L203 ?

I think this should be fine. Maybe an update can be removed as well

**Comment by hedjour** [comment]
Done

**Comment by NicoKiaru** [comment]
Let's get this merged and move on from here.

Thanks again for all the work, and sorry for the super long delay

**Comment by hedjour** [comment]
Thanks

File: install_abba.sh [hunk 8]
```diff
@@ -1,4 +1,6 @@
 #!/bin/bash
+scriptpath=$(realpath $(dirname $0))
+source "$scriptpath/global_function.sh"
```
Comment (hedjour) [inline install_abba.sh:3]: On peut réduire le code ainsi ? qu'en penses tu ?
```suggestion
if [[ "$OSTYPE" == "linux-gnu"* ]] || [[ "$OSTYPE" == "msys" ]]; then
```

Comment (NicoKiaru) [reply install_abba.sh:3]: Alors je ne suis pas sûr de bien comprendre la modif avec le commentaire, mais si c'est plus compact et que ça évite des duplicats de code, oui, impecc! Tu veux ouvrir un PR ? Peut etre que cela sera plus simple pour moi de comprendre la modif.

Comment (hedjour) [reply install_abba.sh:3]: Yep pas de soucis je la fait partout et te fait une PR je t'avoue que je découvre ce genre de fonctionnalité de VSCode

**Pull request merged**